cmake/FindRust.cmake

```diff
@@ -430,6 +430,8 @@ foreach(LINE ${LINE_LIST})
     string(REPLACE "native-static-libs: " "" LINE "${LINE}")
     string(REGEX REPLACE "  " "" LINE "${LINE}")
     string(REGEX REPLACE " " ";" LINE "${LINE}")
+    # remove linker flags
+    list(FILTER LINE EXCLUDE REGEX "/.*")
```

Comment:
I think the LINE is for all the flags.  This works great if the flag we're filtering is at the end (which it has been from what I can tell), but if any real libs were to appear afterwards, I think this would also remove them. 

What do you think about this?
```
    list(FILTER LINE EXCLUDE REGEX "/[a-zA-Z:]*;?")
```

Reply:
Oh, it's filtering `LINE` as a list. In CMake, a list is just a semi-colon separated string.

Reply:
Here is the example: https://godbolt.org/z/bK1M5aabY

Reply:
Oh, that's awesome. ❤️